tests/integration/test_tls.py

```diff
@@ -0,0 +1,383 @@
+        stderr=subprocess.PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+    assert "TLSv1.3" in response
+    assert "CN = kyuubi" in response
+
+    # get issued certificates
+    logger.info("Get certificate from self-signed certificate operator")
+    self_signed_certificate_unit = ops_test.model.applications[
+        charm_versions.tls.application_name
+    ].units[0]
+    action = await self_signed_certificate_unit.run_action(
+        action_name="get-issued-certificates",
+    )
+    result = await action.wait()
+    items = ast.literal_eval(result.results.get("certificates"))
+    certificates = json.loads(items[0])
+    cert = certificates["certificate"]
+
+    logger.info(f"Copy the certificate to the testpod in this location: {CERTIFICATE_LOCATION}")
+    with umask_named_temporary_file(
+        mode="w",
+        prefix="cert-",
+        suffix=".conf",
+        dir=os.path.expanduser("~"),
+    ) as temp_file:
+        with open(temp_file.name, "w+") as f:
+            f.writelines(cert)
+        kubectl_command = [
+            "kubectl",
+            "cp",
+            "-n",
+            ops_test.model_name,
+            temp_file.name,
+            f"{test_pod}:{CERTIFICATE_LOCATION}",
+            "-c",
+            "kyuubi",
+        ]
+        process = subprocess.run(kubectl_command, capture_output=True, check=True)
+        logger.info(process.stdout.decode())
+        logger.info(process.stderr.decode())
+        assert process.returncode == 0
+
+        logger.info(
+            f"Generating the trustore in the testpod in this location: {TRUSTSTORE_LOCATION}"
+        )
+        c2 = [
+            "keytool",
+            "-importcert",
+            "--noprompt",
+            "-alias",
+            "mycert",
+            "-file",
+            CERTIFICATE_LOCATION,
+            "-keystore",
+            TRUSTSTORE_LOCATION,
+            "-storepass",
+            TRUSTSTORE_PASSWORD,
+        ]
+
+    await run_command_in_pod(ops_test, "testpod", c2)
+    # mod permission of the trustore
+    c3 = ["chmod", "u+x", TRUSTSTORE_LOCATION]
+    await run_command_in_pod(ops_test, "testpod", c3)
+
+    # run query with tls
+    logger.info("Running action 'get-jdbc-endpoint' on kyuubi-k8s unit...")
+    kyuubi_unit = ops_test.model.applications[APP_NAME].units[0]
+    action = await kyuubi_unit.run_action(
+        action_name="get-jdbc-endpoint",
+    )
+    result = await action.wait()
+
+    jdbc_endpoint = result.results.get("endpoint")
+    logger.info(f"JDBC endpoint: {jdbc_endpoint}")
+
+    jdbc_endpoint_ssl = (
+        jdbc_endpoint
+        + f";ssl=true;trustStorePassword={TRUSTSTORE_PASSWORD};sslTrustStore={TRUSTSTORE_LOCATION}"
+    )
+    logger.info(f"JDBC endpoint with SSL: {jdbc_endpoint_ssl}")
+
+    logger.info(
+        "Testing JDBC endpoint by connecting with beeline" " and executing a few SQL queries..."
+    )
+    process = subprocess.run(
```

Comment:
suggestion: We can already start using the new `run_sql_test_against_jdbc_endpoint(ops_test, test_pod)` helper function here.